pages/p/[productId].js

```diff
@@ -120,6 +120,22 @@ const Product = React.memo(lazyProps => {
     </Row>
   )
 
+  function fetchVariant() {
+    fetch(`/api/p/${product.id}?color=${color.id}&size=${size.id}`)
+      .then(res => res.json())
+      .then(data => {
+        updateStore({
+          ...store,
+          pageData: { ...store.pageData, ...data },
+        })
+      })
+  }
+
+  // Fetch variant data upon changing color or size options
+  useEffect(() => {
+    fetchVariant()
+  }, [color.id, size.id])
```

Comment:
also page.id

Reply:
I don't think it is needed here since when you link a product page, the product state is updated

Reply:
It's just a good practise to put all the deps which you use inside useEffect, but it's not needed yeap.